torch/_prims/context.py

```diff
@@ -170,13 +151,23 @@ def __torch_function__(
         mapping = torch_to_refs_map()
         func = mapping.get(orig_func, None)
 
+        skip_decomposition_table: Sequence[
+            Union[torch._ops.OpOverload, torch._ops.OpOverloadPacket]
+        ] = {  # type: ignore[assignment]
+            # control-flow dependent decomposition. See Issue #85517
+            torch.ops.aten.masked_fill.Tensor,
+        }
         # For torch.ops.aten.*, use registered decompositions from torch._decomp
         # torch._decomp.decomposition_table provides a mapping from
         # torch.ops.aten.* to torch._refs or torch._decomp.decompositions
         # implementations.
         # There're other ways to implement this functionality,
         # see https://github.com/pytorch/pytorch/pull/82657#discussion_r939776417
-        if func is None and isinstance(orig_func, torch._ops.OpOverload):
+        if (
+            func is None
```

Comment:
This would essentially make `masked_fill` a prim, right? Maybe an nvFuser-specific lowering (or not) for it would be preferable?

Reply:
Thanks for the comment. @IvanYashchuk made similar comment offline.

I'll update this to nvprim code path instead.